### PR TITLE
Curses! This dependency must be removed!

### LINKED ISSUE #3007: Get rid of curses dependency
@melted has found a different, more lightweight way to do what we do with curses: https://hackage.haskell.org/package/terminal-size

### PR DESCRIPTION
**Pull request by melted**
Remove curses dependency

fixes #3007

### CODE REVIEW

File: custom.mk-alldeps [hunk 1]
```diff
@@ -1 +1 @@
-CABALFLAGS += -f LLVM -f FFI -f curses --disable-documentation --disable-profiling --disable-library-profiling
+CABALFLAGS += -f GMP  -f FFI --disable-documentation --disable-profiling --disable-library-profiling
```
Comment (david-christiansen) [inline custom.mk-alldeps]: Nice catch!

**Comment by david-christiansen** [comment]
Looking good! Hopefully this will also eliminate the need for all the screen width flags in the tests to avoid Curses vomiting everywhere. Fun title too!

**Comment by melted** [comment]
The screen width in the tests was fixed in #2974

**Comment by david-christiansen** [comment]
Well then! :-)

**Comment by david-christiansen** [comment]
Thanks so much!

**Pull request merged**